### PR TITLE
AP_Mount: set clock on SIYI

### PR DESCRIPTION
**Pull request by tridge**
this means photos on microSD have correct date, based on time we get from GPS, and allows for easier geo-reference

### CODE REVIEW

**Review by rmackay9 — CHANGES_REQUESTED** [review]
I think it would be better to move this down below the "request firmware" section.  As it is, I think you'll find that the time is only set if the gimbal is powered up at the same time as the autopilot.  If the gimbal is powered up say 30seconds after the autopilot the time will never be set.

Doing this may also remove the need to setting the time 5 times.

**Comment by tridge** [comment]
> I think it would be better to move this down below the "request firmware" section.

thanks, done.
I left in the "5 times" code as we have no way to validate this pkt has gotten through, and it will be annoying if users sometimes get back timestamps on photos

**Review by rmackay9 — APPROVED** [review]

File: libraries/AP_Mount/AP_Mount_Siyi.cpp [hunk 1]
```diff
@@ -68,6 +69,15 @@ void AP_Mount_Siyi::update()
         } else {
             request_configuration();
         }
+
+        // send UTC time to the camera
+        if (_got_hardware_id && sent_time_count < 5) {
```
Comment (rmackay9) [inline libraries/AP_Mount/AP_Mount_Siyi.cpp:74]: I don't think we need the _got_hardware_id here.. it's already checked above and has a return after it.

Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Siyi.cpp:74]: it's an existing problem but we could also remove the "} else {" the precedes the "request_configuration".  The only reason to do this is that it makes it more clear that the request_configuration() and set_time are run together.

**Pull request merged**